### ISSUE TITLE
high cpu since 0.11.1

### ISSUE THREAD

**Issue by lampewebdev** [TabbyML/tabby#2222]
**Describe the bug**
Tabby has high cpu usage randomly.

**Information about your version**
Please provide output of `tabby --version`
tabby 0.11.1

**Information about your GPU**
Apple M3 Metal

**Additional context**
<img width="1182" alt="image" src="https://github.com/TabbyML/tabby/assets/44268468/e077000b-0074-4720-9730-0ecc44840966">

**Comment by wsxiaoys** [comment]
Hi - could you share a bit more about your configuration, including:
1. what's your start command (the screenshot might cut off some of it).
2. have you configured any repository (git / github / gitlab)?

**Comment by lampewebdev** [comment]
> 1. what's your start command (the screenshot might cut off some of it).
```
 tabby serve --port 9090 --device metal --model StarCoder-1B
```
> 2. have you configured any repository (git / github / gitlab)?
no

**Comment by lampewebdev** [comment]
I have now wiped the .tabby and .tabby-agent folders and redownloaded and setup tabby.  

Lets see if this will solve it.

**Comment by wsxiaoys** [comment]
Thank -This issue has also been reported on Slack. however, we have not yet identified a reliable method to reproduce it.

I have been working with the exact same configuration for several days, but I have also been unable to reproduce the issue. 👀

**Comment by lampewebdev** [comment]
Maybe I'm completely wrong but I usually notice it in the evening after the macbook was on a long stand by but this can to coincidence.

**Comment by wsxiaoys** [comment]
I used `top` to sample ~10 hours data, and there's only 7 data points with cpu usage >= 10

<img width="2402" alt="image" src="https://github.com/TabbyML/tabby/assets/388154/1b63b12c-73da-40fa-9e5f-efe35a0ac1c0">

**Comment by lampewebdev** [comment]
let me run glances today and I will come back with my findings. 

Would it help to turn on logs to debug in tabby?

**Comment by lsouchet** [comment]
Same here on a Macbook air M3, the CPU skyrockets randomly after a few hours, usually when I am not even editing code in VS code.
Here is a screenshot on the activity monitor.
<img width="839" alt="image" src="https://github.com/TabbyML/tabby/assets/9589272/6780a379-89af-4721-9788-f5f2f9b0e8ff">

I start the server with `tabby serve --device metal --model DeepseekCoder-1.3B`

I am willing to gather more data, do you have guidance an suggestion about what data would be useful and how to retrieve it?

**Comment by lampewebdev** [comment]
So it hapend again. The logs show nothing. If I see it correctly it started when the macbook was in standby.

**Comment by sheyll** [comment]
I can confirm this behaviour on my M3 too: tabby running and after a few hours the CPU usage goes above 470%

**Comment by wsxiaoys** [comment]
Hi, just trying to summarize the situation (as I still cannot reproduce it for now).
0. Problematic version is 0.11.x
1. this issue seems only happens for metal deployment.
2. CPU usage will go high after tabby running for a few hours.
3. Passing `--no-webserver` seems can work this around (disabling the web ui)

**Comment by wsxiaoys** [comment]
seems related to apalis sqlite polling - but cannot confirm since i'm not able to reduce. sending #2246 to fix anyway

**Comment by wsxiaoys** [comment]
The issue seems go away in 0.12.0-rc.0

**Comment by tups** [comment]
I see the same performance issues, on version 0.12.0-rc1, I did not have this problem on a previous version (0.7.0).
My battery is melting like snow in the sun since I updated to 0.11

<img width="837" alt="Capture d’écran 2024-05-31 à 23 09 16" src="https://github.com/TabbyML/tabby/assets/868611/75fb6369-92c8-43d8-9e6b-afe21e0f6c29">


<img width="707" alt="Capture d’écran 2024-05-31 à 23 11 12" src="https://github.com/TabbyML/tabby/assets/868611/eddc7dd4-edfe-48d2-a637-65ede8b59db4">

It's cool if you find where it comes from

**Comment by tups** [comment]
Finally after several hours of working, it returned to normal with 0.12.0-rc.1 on my side:
<img width="1102" alt="Capture d’écran 2024-06-03 à 08 51 37" src="https://github.com/TabbyML/tabby/assets/868611/75be5ef5-9288-4a14-befe-a25a9cad6230">

**Comment by wsxiaoys** [comment]
fixed in 0.12 release: https://github.com/TabbyML/tabby/releases/tag/v0.12.0
